### ISSUE TITLE
[googlefonts] canonical_stylename condition doesn't work for VFs, possibly more

### ISSUE THREAD

**Issue by yanone** [fonttools/fontbakery#4178]
It appears like the GF profile has never been thoroughly updated after VFs were introduced. After I had already fixed at least one condition in the past, here's another one that just quietly fails when it's ran on VFs: `canonical_stylename`.

The problem with the quiet failing is that it's quiet. There's no way for a user to know whether the check has run or whether it was skipped erroneously, so we just keep shipping faulty fonts.

I’m writing this issue to plead that the entire GF codebase gets reviewed for VFs and that each condition gets its own code test to be able to handle VFs reliably.

**Comment by yanone** [comment]
Here's a general path forward for removing non-condition methods from each check's conditions:

Integrate into the condition runner a condition checking for whether or not a condition's name start with "is_" or "are_". If they aren't, throw an error.

Then take all those non-conditions out of the check's parameter interface and instead import and invoke them normally in the check's code.

So 

```python
@check(
    conditions=["are_ttf", "canonical_stylename"],
)
def com_google_fonts_check_family_equal_numbers_of_glyphs(ttFonts, canonical_stylename):
```

becomes:

```python
@check(
    conditions=["are_ttf"],
)
def com_google_fonts_check_family_equal_numbers_of_glyphs(ttFonts):
    from .googlefonts_conditions import canonical_stylename
    canonical_stylename = canonical_stylename(ttFonts)
```

The above example is actually slightly made-up. The respective check is deactivated since four years. But I'm trying to mitigate the entirety of normal methods being (ab)used as condition.

**Comment by graphicore** [comment]
But, this (your second "becomes" example):

```py
@check(
    conditions=["are_ttf"],
)
def com_google_fonts_check_family_equal_numbers_of_glyphs(ttFonts):
    from .googlefonts_conditions import canonical_stylename
    canonical_stylename = canonical_stylename(ttFonts)
    # use canonical_stylename
```

equals

```py
@check(
    conditions=["are_ttf"],
)
def com_google_fonts_check_family_equal_numbers_of_glyphs(ttFonts, canonical_stylename):
    # use canonical_stylename
```

> Then take all those non-conditions out of the check's parameter interface and instead import and invoke them normally in the check's code.

There's no reason to do this. The parameter injection and the `@checks conditions` don't depend on each other. It seems like you expect the parameter `canonical_stylename` is only available if it is used in `@check conditios`, but this is not the case.

**Comment by graphicore** [comment]
I don't agree with a lot of that, but I also think it's off-topic and I would prefer to discuss it in another issue.
